model/countries/AU/AU-parsing-rules.yaml

```diff
@@ -0,0 +1,232 @@
+          capture_reference: ParseUnit
+      - extract_part:
+          capture_reference: ParseFloor
+
+  unit:
+    decomposition:
+      capture:
+        output: unit
+        parts:
+        - capture:
+            output: unit-type
+            parts: [ {regex_reference: kUnitTypeLiteralRe} ]
+            quantifier: MATCH_OPTIONAL
+        - separator: {regex_reference: kWhitespaceSeparator}
+        - capture:
+            output: unit-name
+            parts: [ {regex_reference: kUnitNameValueRe} ]
+
+
+test_regex_definitions:
+- id: "kUnitTypeLiteralRe: Apt."
+  regex_name: kUnitTypeLiteralRe
+  input: "Apt."
+  match_groups: ["Apt."]
+- id: "kUnitTypeLiteralRe: APT"
+  regex_name: kUnitTypeLiteralRe
+  input: "APT"
+  match_groups: ["APT"]
+- id: "kUnitTypeLiteralRe: Suite"
+  regex_name: kUnitTypeLiteralRe
+  input: "Suite"
+  match_groups: ["Suite"]
+- id: "kUnitTypeLiteralRe: STE"
+  regex_name: kUnitTypeLiteralRe
+  input: "STE"
+  match_groups: ["STE"]
+- id: "kUnitTypeLiteralRe: ste."
+  regex_name: kUnitTypeLiteralRe
+  input: "ste."
+  match_groups: ["ste."]
+- id: "kUnitTypeLiteralRe: SE"
+  regex_name: kUnitTypeLiteralRe
+  input: "SE"
+  match_groups: ["SE"]
+- id: "kUnitTypeLiteralRe: Unit"
+  regex_name: kUnitTypeLiteralRe
+  input: "Unit"
+  match_groups: ["Unit"]
+- id: "kUnitTypeLiteralRe: U"
+  regex_name: kUnitTypeLiteralRe
+  input: "U"
+  match_groups: ["U"]
+- id: "kUnitTypeLiteralRe: rm"
+  regex_name: kUnitTypeLiteralRe
+  input: "rm"
+  match_groups: ["rm"]
+- id: "kUnitTypeLiteralRe: room"
+  regex_name: kUnitTypeLiteralRe
+  input: "room"
+  match_groups: ["room"]
+- id: "kUnitTypeLiteralRe: PO box"
+  regex_name: kUnitTypeLiteralRe
+  input: "PO box"
+  match_groups: ["PO box"]
+- id: "kFloorLiteralRe: ug"
+  regex_name: kFloorLiteralRe
+  input: "ug"
+  match_groups: ["ug"]
+
+    
+
+test_parsing_definitions:
+- id: "Test 1"
+  type: unit
+  input: "Apartment 75"
+  output:
+    unit: "Apartment 75"
+    unit-type: "Apartment"
+    unit-name: "75"
+- id: "Test 2"
+  type: in-building-location
+  input: "Apt. 75 Level 7"
+  output:
+    unit: "Apt. 75"
+    unit-type: "Apt."
+    unit-name: "75"
+    floor: "7"
+- id: "Test 2"
+  type: in-building-location
+  input: "Apt. 75 Floor 7"
+  output:
+    unit: "Apt. 75"
+    unit-type: "Apt."
+    unit-name: "75"
+    floor: "7"
+- id: "Test 3"
+  type: in-building-location
+  input: "suite 75 ug"
+  output:
+    unit: "suite 75"
+    unit-type: "suite"
+    unit-name: "75"
+- id: "Test 4"
+  type: building-location
+  input: "75 Lane Street"
+  output:
+    street: "Lane Street"
+    building: "75"
+- id: "Test 5"
+  type: building-location
+  input: "189 Great Eastern Highway"
+  output:
+    street: "Great Eastern Highway"
+    building: "189"
+- id: "Test 6"
+  type: building-location
+  input: "189 Great Eastern Highway"
+  output:
+    street: "Great Eastern Highway"
+    building: "189"
+- id: "Test 7"
+  type: street-address-alternative-1
```

Comment:
I think we should handle also cases where the unit, floor (or both?) are missing. It's not clear to me that under the current setup that's possible.

Reply:
That's the new tests that I'm adding (still in process)

Reply:
Done